test/config/test_configurable.rb

```diff
@@ -604,93 +683,140 @@ def e(name, arg = '', attrs = {}, elements = [])
+            ConfigurableSpec::Overwrite::Multi.new.configure(CONF1)
+          end
+        end
+
+        test 'subclass cannot overwrite alias' do
+          assert_raise(Fluent::ConfigError.new("subclass cannot overwrite base class's definition")) do
+            ConfigurableSpec::Overwrite::Alias.new.configure(CONF1)
+          end
+        end
+
+        test 'subclass uses superclass default options' do
+          base = ConfigurableSpec::Overwrite::Base.new.configure(CONF2)
+          sub = ConfigurableSpec::Overwrite::DefaultOptions.new.configure(CONF2)
+          detail_base = base.class.merged_configure_proxy.sections[:detail]
+          detail_sub = sub.class.merged_configure_proxy.sections[:detail]
+          detail_base_attributes = {
+            requried: detail_base.required,
+            multi: detail_base.multi,
+            alias: detail_base.alias,
+          }
+          detail_sub_attributes = {
+            requried: detail_sub.required,
+            multi: detail_sub.multi,
+            alias: detail_sub.alias,
+          }
+          assert_equal(detail_base_attributes, detail_sub_attributes)
+        end
+
+        test 'subclass can overwrite detail.address' do
+          base = ConfigurableSpec::Overwrite::Base.new.configure(CONF2)
+          target = ConfigurableSpec::Overwrite::DetailAddressDefault.new.configure(CONF2)
+          expected_addresses = ["x", "y"]
+          actual_addresses = [base.detail.address, target.detail.address]
+          assert_equal(expected_addresses, actual_addresses)
+        end
+
+        test 'subclass can add param' do
+          assert_raise(Fluent::ConfigError.new("'phone_no' parameter is required, in section detail")) do
+            ConfigurableSpec::Overwrite::AddParam.new.configure(CONF3)
+          end
+          ex9 = ConfigurableSpec::Overwrite::AddParam.new.configure(CONF4)
+          expected ={
+            address: "Chiyoda Tokyo Japan",
+            phone_no: "+81-00-0000-0000"
+          }
+          actual = {
+            address: ex9.detail.address,
+            phone_no: ex9.detail.phone_no
+          }
+          assert_equal(expected, actual)
+        end
+
+        sub_test_case 'final' do
+          CONF = config_element('ROOT', '', {},
+                                [config_element('appendix', '', {"phone_no" => "+81-0000-0000"}, [])])
+          test 'subclass can overwrite appendix.name, appendix.age w/ level 2' do
+            finalized = ConfigurableSpec::Final::Finalized.new
+            finalized.configure(CONF)
+            expected = { name: "y", age: 10 }
+            actual = { name: finalized.appendix.name, age: finalized.appendix.age }
+            assert_equal(expected, actual)
+          end
+
+          test 'subclass cannot overwrite appendix.name, appendix.age w/ level 3' do
+            level3 = ConfigurableSpec::Final::InheritsFinalized.new
```

Comment:
Is it intentional not to raise exceptions for overwriting required/multi/alias onto finalized subsections, besides to raise exceptions for overwriting these onto non-finalized subsections?